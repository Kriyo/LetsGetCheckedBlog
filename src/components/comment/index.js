import React from 'react'

import {
  Container,
  Item,
  RootItem,
  NestedItem,
  Count,
  Image,
  DetailsWrapper,
  Content,
  AuthorDetails,
  ReplyHeading,
  Form,
  Button,
  ButtonWrapper,
  ErrorLabel,
  Input,
  TextArea,
  Section,
} from './styles/comment'

export const Comment = ({ children, ...props }) => {
  return <Item {...props}>{children}</Item>
}

Comment.Frame = ({ children, ...props }) => {
  return <Container {...props}>{children}</Container>
}

Comment.Section = ({ children, ...props }) => {
  return <Section {...props}>{children}</Section>
}

Comment.RootComment = ({ children, ...props }) => {
  return <RootItem {...props}>{children}</RootItem>
}

Comment.NestedComment = ({ children, ...props }) => {
  return <NestedItem {...props}>{children}</NestedItem>
}

Comment.Count = ({ children, ...props }) => {
  return <Count {...props}>{children}</Count>
}

Comment.DetailsWrapper = ({ children, ...props }) => {
  return <DetailsWrapper {...props}>{children}</DetailsWrapper>
}

Comment.AuthorDetails = ({ children, ...props }) => {
  return <AuthorDetails {...props}>{children}</AuthorDetails>
}

Comment.Avatar = ({ children, ...props }) => {
  return <Image {...props}>{children}</Image>
}

Comment.Content = ({ children, ...props }) => {
  return <Content {...props}>{children}</Content>
}

Comment.ReplyHeading = ({ children, ...props }) => {
  return <ReplyHeading {...props}>{children}</ReplyHeading>
}

Comment.Form = ({ children, ...props }) => {
  return <Form {...props}>{children}</Form>
}

Comment.ButtonWrapper = ({ children, ...props }) => {
  return <ButtonWrapper {...props}>{children}</ButtonWrapper>
}

Comment.Button = ({ children, ...props }) => {
  return <Button {...props}>{children}</Button>
}
Comment.ErrorLabel = ({ children, ...props }) => {
  return <ErrorLabel {...props}>{children}</ErrorLabel>
}
Comment.UserInput = ({ children, ...props }) => {
  return <Input {...props}>{children}</Input>
}

Comment.TextArea = ({ children, ...props }) => {
  return <TextArea {...props}>{children}</TextArea>
}
