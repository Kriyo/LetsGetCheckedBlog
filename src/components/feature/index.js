import React from 'react'

import { Container, Title, SubTitle } from './styles/feature'

export const Feature = ({ children, ...props }) => {
  return <Container {...props}>{children}</Container>
}

Feature.Title = ({ children, ...props }) => {
  return <Title {...props}>{children}</Title>
}

Feature.SubTitle = ({ children, ...props }) => {
  return <SubTitle {...props}>{children}</SubTitle>
}
