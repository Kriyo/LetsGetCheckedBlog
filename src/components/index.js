export { Comment } from './comment'
export { Feature } from './feature'
export { NavBar } from './navbar'
export { Post } from './post'
