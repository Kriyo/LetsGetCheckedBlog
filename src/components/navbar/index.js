import React from 'react'
import { Background, Container, Link, Logo } from './styles/navbar'

export default function NavBar({ background = true, children, ...props }) {
  return background ? <Background {...props}>{children}</Background> : children
}

NavBar.Frame = function NavBarFrame({ children, ...props }) {
  return <Container {...props}>{children}</Container>
}

NavBar.Link = function NavBarLink({ children, ...props }) {
  return <Link {...props}>{children}</Link>
}

NavBar.Logo = function NavBarLogo({ to, ...restProps }) {
  return <Logo {...restProps} />
}
