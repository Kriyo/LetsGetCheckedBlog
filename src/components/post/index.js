import React from 'react'

import {
  Container,
  Item,
  Author,
  Title,
  Description,
  PublishDate,
  Content,
} from './styles/post'

export const Post = ({ children, ...props }) => {
  return <Item {...props}>{children}</Item>
}

Post.Frame = ({ children, ...props }) => {
  return <Container {...props}>{children}</Container>
}

Post.Author = ({ children, ...props }) => {
  return <Author {...props}>{children}</Author>
}

Post.Title = ({ children, ...props }) => {
  return <Title {...props}>{children}</Title>
}

Post.Description = ({ children, ...props }) => {
  return <Description {...props}>{children}</Description>
}

Post.PublishDate = ({ children, ...props }) => {
  return <PublishDate {...props}>{children}</PublishDate>
}

Post.Content = ({ children, ...props }) => {
  return <Content {...props}>{children}</Content>
}
