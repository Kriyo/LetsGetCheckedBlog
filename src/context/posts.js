import { createContext } from 'react'

export const PostsContext = createContext({
  blog: null,
  error: null,
  loading: true,
})
