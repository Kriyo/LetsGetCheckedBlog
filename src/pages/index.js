export { Home } from './home'
export { Blog } from './blog'
export { BlogPost } from './blogPost'
export { About } from './about'
export { Contact } from './contact'
