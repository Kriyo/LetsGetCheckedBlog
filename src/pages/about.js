import React from 'react'

import { Feature } from '../components'
import { NavContainer } from '../containers'

export const About = () => {
  return (
    <>
      <NavContainer>
        <Feature>
          <Feature.Title>About Me</Feature.Title>
          <Feature.SubTitle>
            Et harum quidem rerum facilis est et expedita distinctio. Nam
            liberotempore.
          </Feature.SubTitle>
        </Feature>
      </NavContainer>
    </>
  )
}
