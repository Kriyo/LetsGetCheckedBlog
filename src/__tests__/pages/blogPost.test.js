import React from 'react'
import { render } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom'
import { rest } from 'msw'
import { setupServer } from 'msw/node'
import { BlogPost } from '../../pages'
import { Comment } from '../../components'
import { PostsContext } from '../../context/posts'
import {
  HOME_LOADING_STATE_CONTEXT_PROVIDER,
  HOME_ERROR_STATE_CONTEXT_PROVIDER,
  HOME_WORKING_STATE_CONTEXT_PROVIDER,
  MOCK_COMMENTS_RESPONSE,
  MOCK_SINGLE_POST_URL,
} from '../../constants/fixtures'

// Simulate API requests via mocks
const server = setupServer(
  rest.get('/posts', (req, res, ctx) => {
    return res(ctx.json(MOCK_COMMENTS_RESPONSE))
  })
)

// Simulate URL for blog post slug
Object.defineProperty(window, 'location', {
  value: new URL(MOCK_SINGLE_POST_URL),
  writable: true,
})

beforeAll(() => server.listen())
afterEach(() => server.resetHandlers())
afterAll(() => server.close())

test('renders the <BlogPost/> in a loading state/>', () => {
  const { getByText } = render(
    <Router>
      <PostsContext.Provider value={HOME_LOADING_STATE_CONTEXT_PROVIDER}>
        <BlogPost />
      </PostsContext.Provider>
    </Router>
  )

  expect(getByText('Loading Blog Post')).toBeTruthy()
})

test('renders the <Blog/> in an error state/>', () => {
  const { getByText } = render(
    <Router>
      <PostsContext.Provider value={HOME_ERROR_STATE_CONTEXT_PROVIDER}>
        <BlogPost />
      </PostsContext.Provider>
    </Router>
  )

  expect(getByText('Error Loading Blog Post')).toBeTruthy()
})

test('renders the <Blog/> in a working state/>', () => {
  const { container, getByText } = render(
    <Router>
      <PostsContext.Provider value={HOME_WORKING_STATE_CONTEXT_PROVIDER}>
        <BlogPost />
      </PostsContext.Provider>
    </Router>
  )

  expect(getByText('Blog post #1')).toBeTruthy()
  expect(container.firstChild).toMatchSnapshot()
})

test('renders the <Comment/> in a loading state/>', () => {
  const { container } = render(
    <Router>
      <PostsContext.Provider value={HOME_WORKING_STATE_CONTEXT_PROVIDER}>
        <Comment />
      </PostsContext.Provider>
    </Router>
  )

  expect(container.firstChild).toMatchSnapshot()
})
