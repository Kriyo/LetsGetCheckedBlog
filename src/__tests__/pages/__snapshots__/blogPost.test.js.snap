// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the <Blog/> in a working state/> 1`] = `
<div
  class="navbar__Background-sc-1waf4cx-0 fkfKQq"
  data-testid="nav-bg"
>
  <nav
    class="navbar__Container-sc-1waf4cx-1 iTnWlJ"
  >
    <a
      href="/"
    >
      <img
        alt="Lets Get Checked Blog Logo"
        class="navbar__Logo-sc-1waf4cx-3 dKtmPt"
        src="white-logo.png"
      />
    </a>
    <a
      alt="Go to Home page"
      class="navbar__NavLink-sc-1waf4cx-2 nBHVE"
      href="/"
    >
      Home
    </a>
    <a
      alt="Go to Blog page"
      class="navbar__NavLink-sc-1waf4cx-2 nBHVE"
      href="/blog"
    >
      Blog
    </a>
    <a
      alt="Go to Contact page"
      class="navbar__NavLink-sc-1waf4cx-2 nBHVE"
      href="/contact"
    >
      Contact
    </a>
    <a
      alt="Go to About page"
      class="navbar__NavLink-sc-1waf4cx-2 nBHVE"
      href="/about"
    >
      About
    </a>
  </nav>
</div>
`;

exports[`renders the <Comment/> in a loading state/> 1`] = `
<div
  class="comment__Item-sc-1vl0att-1"
/>
`;
